tests/test_eval_jaxpr.py

```diff
@@ -80,21 +83,24 @@ def f(x):
     dev = qml.device(device_name, wires=1)
     jaxpr = jax.make_jaxpr(f)(0.5)
 
-    res = dev.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, 0.5)
+    if use_jit:
+        res = jax.jit(partial(dev.eval_jaxpr, jaxpr.jaxpr))(jaxpr.consts, 0.5)
+    else:
+        res = dev.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, 0.5)
     assert qml.math.allclose(res, jax.numpy.cos(0.5))
 
 
 def test_capture_remains_enabled_if_measurement_error():
     """Test that capture remains enabled if there is a measurement error."""
 
-    dev = qml.device(device_name, wires=1)
+    dev = qml.device(device_name, wires=1, shots=1)
 
     def g():
-        return qml.sample(wires=0)  # sampling with analytic execution.
```

Comment:
Is `qml.sample` also supported with `jax.jit`?

Reply:
Reverting to the no shots + sample case.  Probably changed it when debugging the change in error type.

The point of this test it to trigger some kind of error, and state with finite shots or analytic sampling are equivalent ways to generate an error.

Reply:
OK now I remember why I switched. With sample and no shots, we get an error when we try and calculate the shapes. 

Here I want to test what happens when an error is thrown mid-execution. So I had to alter what was causing an error.